src/System.IO.Pipelines/tests/PipeWriterStreamTests.nonnetstandard.cs

```diff
@@ -167,6 +167,15 @@ public async Task DefaultPipeWriterImplementationReturnsPipeWriterStream()
             Assert.True(pipeWriter.FlushCalled);
         }
 
+        [Fact]
+        public void AsStreamDoNotCompleteWriter()
+        {
+            var pipeWriter = new TestPipeWriter();
+
+            // will throw in Complete if it was actually invoked
+            pipeWriter.AsStream(leaveOpen: true).Dispose();
```

Comment:
In addition to the claim of the above comment, can we actually validate that the pipe writer is still usable / not completed?

Reply:
(Same for the reader case)

Reply:
I have updated this test with PipeWriter implementation which actually throws exception on any method call. And PipeWriter/PipeReader basic contract doesn't provide any method or property like IsCompleted. 
So this test effectively asserts no method would be called in PipeWriter/Reader.